libraries/ms-drivers/src/ads1259_adc.c

```diff
@@ -1,4 +1,5 @@
 #include "ads1259_adc.h"
```

Comment:
A few things I noticed about this driver:
* `Ads1259ErrorHandlerCb` has a `context` parameter, but there's no `context` field in the settings, and we just pass `NULL` for the context always, so the parameter is useless.
* We initialize SPI inside `ads1259_init`; I'd prefer to initialize it externally to the module like normal for better interoperability.
* The parameters are in the wrong order in `ads1259_init`, it's conventionally `storage, settings` to put storage first always.
* There's no callback function for when a conversion is ready, so a client (like the smoke test) has to wait with its own soft timer for the conversion time, which is really an implementation detail of the ADS1259 driver.

You can fix these if you'd like, though you don't have to if there's any danger it would break hardware validation.

Reply:
- Added context field in settings
- There's actually not much convention on whether to init spi inside or outside the driver, see AFE and mcp2515 code. I'll leave it this way for now. (I notice we tend to init spi inside drivers, but i2c outside, may make sense since we rarely [except spv1020] use multiple spi ICs on the same bus but tend to use multiple i2c ICs on the same bus)
- Switched order for init args
- The lack of callback function was a (not necessarily good) design decision since it was a concern that the IC would hog processing time if it interrupted every cycle. It's much higher performance than we're using, so this way we can more easily tune it to whatever rate is necessary.